### ISSUE TITLE
Regression: static methods 

### ISSUE THREAD

**Issue by cevek** [rollup/rollup#1362]
Rollup version: 0.41.6
A.js
```js
import {B} from './B';
export function A() {
    B();
}
```

B.js
```js
import {C} from './C';
export function B() {}

function BB() {
    new C();
}
```

C.js
```js
export var C = function () {}
C.foo = 1;
```

`rollup --format=cjs  --output=out.js A.js`

out.js
```js
'use ';

Object.defineProperty(exports, '__esModule', { value: true });

// C is not used here
var C = function () {};
C.foo = 1;

function B() {}

function A() {
    B();
}

exports.A = A;
```

Previous 0.25 version works

**Comment by BenoitZugmeyer** [comment]
I've got the same issue, with `propTypes` (static properties added to classes).

**Comment by AllNamesRTaken** [comment]
Even easier. Any typescript written class with static properties will automatically fail to treeshake. This ruins my day :(.
[rollup repl example](https://rollupjs.org/repl?version=0.41.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGZ29vZGNvcmUuanMnJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmdvb2Rjb3JlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMnZhciUyMEZvbyUzRCUyMChmdW5jdGlvbiUyMCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGZ1bmN0aW9uJTIwRm9vKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMEZvbyUzQiU1Q24lN0QoKSklM0IlNUNuRm9vLnN0YXRpY1Byb3AlMjAlM0QlMjAlNUMlMjJzb21ldmFsdWUlNUMlMjIlM0IlNUNuJTVDbmV4cG9ydCUyMCU3QiUyMEZvbyUyMCU3RCUzQiU1Q24lMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybW9kdWxlTmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMm1vZHVsZUlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

**Comment by AllNamesRTaken** [comment]
Also this causes any exported custom object such as:
```
var TheInstance = new Foo();
export { TheInstance };
```
to cause Foo to get included even when noone is referencing TheInstance or Foo.

While doing this with native Object work as intended.
```
var OtherInstance = new Object();
export { OtherInstance };
```

**Comment by lukastaegert** [comment]
It seems this is resolved in v0.49.2:
[REPL](https://rollupjs.org/repl?version=0.49.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QkIlN0QlMjBmcm9tJTIwJy4lMkZCJyUzQiU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMEEoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjBCKCklM0IlNUNuJTdEJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMkIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwJTdCQyU3RCUyMGZyb20lMjAnLiUyRkMnJTNCJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIwQigpJTIwJTdCJTdEJTVDbiU1Q25mdW5jdGlvbiUyMEJCKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwbmV3JTIwQygpJTNCJTVDbiU3RCUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJDLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMHZhciUyMEMlMjAlM0QlMjBmdW5jdGlvbiUyMCgpJTIwJTdCJTdEJTVDbkMuZm9vJTIwJTNEJTIwMSUzQiUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

Close the issue?
